### PR TITLE
remove DefaultCookieDecoder

### PR DESCRIPTION
**Pull request by n0tl3ss**
#10435 remove DefaultCookieDecoder from services because its broken for parsing request cookies

### CODE REVIEW

**Review by yawkat — APPROVED** [review]

**Comment by sdelamo** [comment]
should not we fix it if it is broken?

**Comment by sdelamo** [comment]
if it is bug it should have 4.3.x as base.

**Comment by yawkat** [comment]
this is related to https://github.com/micronaut-projects/micronaut-core/issues/10435 . we have to fix it properly later, this PR just fixes the precedence of the netty decoder

**Comment by sdelamo** [comment]
I created an alternative fix https://github.com/micronaut-projects/micronaut-core/pull/10661 which allows users to override the implementation.

**Review by sdelamo — APPROVED** [review]
I have removed the following:

http/src/main/resources/META-INF/services/io.micronaut.http.cookie.ClientCookieEncoder
http/src/main/resources/META-INF/services/io.micronaut.http.cookie.ServerCookieDecoder
http/src/main/resources/META-INF/services/io.micronaut.http.cookie.ServerCookieEncoder

and added tests. 

@yawkat check it out.

File: http-netty/src/test/groovy/io/micronaut/http/netty/cookies/NettyLaxClientCookieEncoderSpec.groovy [hunk 1]
```diff
@@ -1,8 +1,9 @@
 package io.micronaut.http.netty.cookies
 
+import io.micronaut.core.order.OrderUtil
```
Comment (sdelamo) [inline http-netty/src/test/groovy/io/micronaut/http/netty/cookies/NettyLaxClientCookieEncoderSpec.groovy]:
```suggestion
```

File: http-netty/src/test/groovy/io/micronaut/http/netty/cookies/NettyLaxClientCookieEncoderSpec.groovy [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
 import io.micronaut.http.cookie.ClientCookieEncoder
 import io.micronaut.http.cookie.Cookie
-
+import io.micronaut.http.cookie.DefaultClientCookieEncoder
```
Comment (sdelamo) [inline http-netty/src/test/groovy/io/micronaut/http/netty/cookies/NettyLaxClientCookieEncoderSpec.groovy]:
```suggestion
```

File: http-netty/src/test/groovy/io/micronaut/http/netty/cookies/NettyServerCookieEncoderSpec.groovy [hunk 3]
```diff
@@ -1,6 +1,8 @@
 package io.micronaut.http.netty.cookies
 
+import io.micronaut.core.order.OrderUtil
 import io.micronaut.http.cookie.Cookie
+import io.micronaut.http.cookie.DefaultServerCookieEncoder
```
Comment (sdelamo) [inline http-netty/src/test/groovy/io/micronaut/http/netty/cookies/NettyServerCookieEncoderSpec.groovy]:
```suggestion
```

File: http-netty/src/test/groovy/io/micronaut/http/netty/cookies/NettyServerCookieEncoderSpec.groovy [hunk 4]
```diff
@@ -1,6 +1,8 @@
 package io.micronaut.http.netty.cookies
 
+import io.micronaut.core.order.OrderUtil
```
Comment (sdelamo) [inline http-netty/src/test/groovy/io/micronaut/http/netty/cookies/NettyServerCookieEncoderSpec.groovy]:
```suggestion
```

**Pull request merged**